pkg/provisioner/controller.go

```diff
@@ -380,7 +391,10 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	// Fill in basic necessary object bucket info
 	setBasicOBInfo := func() {
 		setObjectBucketName(ob, key)
-		ob.Spec.StorageClassName = obc.Spec.StorageClassName
+		// obc can be nil if functions that return a modified version of it fail
```

Comment:
why? If a func the returns an obc fails why does the code continue with that obc? err != nil should returns an error and stop the current func.

Reply:
The code that continues is in the deferred cleanup function.